numpy/core/arrayprint.py

```diff
@@ -1637,12 +1668,96 @@ def array_str(a, max_line_width=None, precision=None, suppress_small=None):
         a, max_line_width, precision, suppress_small)
 
 
+_FORMAT_SPEC_REGEXP = re.compile(
+    r"([ +-])?"        # sign options
+    r"(\.)?([0-9]+)?"  # field precision
+    r"(.)?"            # field type
+)
+
+def _parse_format_spec(fs):
+    """
+    Parse the format spec and returns a dictionary with options 
+    for using it with `array2string`. This is inspired in the 
+    [Format Specification Mini-Language](https://docs.python.org/3/library/string.html#formatspec)
+
+    Parameters
+    ----------
+    fs: str
+        a string with the format specification, e.g., "+.2f"
+
+    Semantic for format spec
+    ------------------------
+
+    format_spec ::=  [sign][.precision][type]
+    sign        ::=  "+" | "-" | " "
+    precision   ::=  [0-9]+
+    type        ::=  "f" | "e" | "g"
+    """
+
+    match = _FORMAT_SPEC_REGEXP.fullmatch(fs)
+    if match is None:
+        raise ValueError("Invalid format specifier")
+
+    sign, period, precision, fmt_code = match.groups()
+
+    options = {}
+
+    if sign is not None:
+        options["sign"] = sign
+
+    if period is not None:
+        if precision is None:
+            raise ValueError("Format specifier missing precision")
+        options["precision"] = int(precision)
+
+    if fmt_code is not None:
+        if fmt_code not in "feg":
+            raise ValueError(f"Unknown format code {fmt_code!r}")
+        if fmt_code == "f":  # try to force fixed precision
+            options["suppress_small"] = True
+        elif fmt_code == "e":  # force exp_format
+            options["exp_format"] = True
+        elif fmt_code == "g":
+            # TODO: implement code in `FloatingFormat` such that
+            # the tests in test_multiarray.py:test_general_format pass
+            # note: remember to add the tests to `test_arrayprint.py:test_type`
+            pass
```

Comment:
Can we either implement this or just make it an error here? Ignoring the format code in this path seems not helpful?